### PR TITLE
Top 5 customers admin merchant jr

### PR DESCRIPTION
**Pull request by jimriddle1**
Hey eveyrone,

Zach and I were working on both of our Advanced AR user stories, and this PR handles both. 

We did get feedback from Dani that our method looks good, even though the result on the rails s was not what I expected.

Let me know if you have any questions.

Thanks,

### CODE REVIEW

File: app/controllers/admins_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class AdminsController < ApplicationController
   def dashboard
+    @customers = Customer.all
```
Comment (z-prince) [inline app/controllers/admins_controller.rb:3]: Great work here Jim, you've come a long way!

File: app/models/customer.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
 class Customer < ApplicationRecord
   has_many :invoices
+  has_many :transactions, through: :invoices
+
+  def self.top_five_by_transaction_count
```
Comment (z-prince) [inline app/models/customer.rb:5]: I feel like you really did a number on this one. Almost two advanced AR stories out of the way.

File: app/models/item.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@ def item_best_day
             .order(sales: :desc)
             .first.updated_at
   end
-
+
```
Comment (z-prince) [inline app/models/item.rb:28]: Not sure what's going on here.

File: app/models/merchant.rb [hunk 4]
```diff
@@ -24,6 +24,16 @@ def top_five_items_by_revenue
          .limit(5)
   end
 
+  def top_five_customers
+     customers.select("customers.*, count(transactions) as total_transaction_count")
+        .where(transactions: {result: "success"})
+         .joins(:transactions)
```
Comment (z-prince) [inline app/models/merchant.rb:30]: This is crucial.

**Review by z-prince — APPROVED** [review]

File: app/models/item.rb (hunk 3 above)
Comment (jimriddle1) [reply app/models/item.rb:28]: strange issue with the git diff it would seem!

**Pull request merged**